tests/test_modeling_common.py

```diff
@@ -2736,6 +2736,58 @@ def test_inputs_embeds(self):
             with torch.no_grad():
                 model(**inputs)[0]
 
+    def test_inputs_embeds_matches_input_ids(self):
+        config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
+
+        for model_class in self.all_model_classes:
+            if model_class.__name__ not in get_values(MODEL_MAPPING_NAMES):
+                continue
+            model = model_class(config)
+            model.to(torch_device)
+            model.eval()
+
+            model_forward_args = inspect.signature(model.forward).parameters
+            if "inputs_embeds" not in model_forward_args:
+                self.skipTest("This model doesn't use `inputs_embeds`")
+
+            inputs = copy.deepcopy(self._prepare_for_class(inputs_dict, model_class))
+            pad_token_id = config.pad_token_id if config.pad_token_id is not None else 1
+            print(inputs.keys())
+
+            try:
```

Comment:
We shouldn't have try/except patterns in tests: either we're deliberately triggering an error or we're not. Raising exceptions is for within code, allowing us to elegantly handle errors at runtime. Instead, models which don't use `input_embeds` should explicitly skip this test, and the try/except block be removed

Reply:
Hmm I thought we are adding all the NotImplementedErrors to get rid of many copies "skipTests", as discussed with @gante above:

> IMO -- instead of adding custom test code to handle a silent failure, let's properly handle the failure (if the user passes inputs_embeds to forward, raise a NotImplementedError, which can be caught in the test)

I can bring back all skips if that's needed for tests-consistency but if we are specifically catching only "NotImplementedErrors" isn't it okay?

Reply:
Ah, OK, sorry, I missed the bit about handling in the tests. 

We should raise NotImplementedError on the model side, but let users handle that however they want, and then explicitly skip in the tests using `unittest.skip` for the specific models. This avoids accidentally skipping because a different NotImplementedError is raised. It's true we end up with more code, [but it's better for tests to be DAMP](https://testing.googleblog.com/2019/12/testing-on-toilet-tests-too-dry-make.html) i.e. very clear and explicit.

Reply:
> [but it's better for tests to be DAMP](https://testing.googleblog.com/2019/12/testing-on-toilet-tests-too-dry-make.html)

hehe the opposite of what I've been telling @zucchini-nlp in this PR 🙈  By bad :D

Reply:
Ah, okay, TIL about the DAMP principle